src/script/interpreter.cpp

```diff
@@ -1100,13 +1105,17 @@ class CTransactionSignatureSerializer {
             const CTransaction& txTo = dynamic_cast<const CTransaction&>(txBaseTo);
 
             // Serialize vin
+#if 0
             // In case of SIGHASH_ANYONECANPAY:
             // * if Tx has Sc support version and nIn belongs to the CSW inputs - skip inputs serialization
             // * otherwise only the input being signed is serialized
             // Otherwise - serialize all inputs
             unsigned int nInputs = fAnyoneCanPay ?
                         (txTo.IsScVersion() && nIn >= txTo.GetVin().size() ? 0 : 1) :
                         txTo.GetVin().size();
+#else
+            unsigned int nInputs = fAnyoneCanPay ? 1 : txTo.GetVin().size();
```

Comment:
For this exactly case, seems better to keep logic as before, to make it independent to the checks outside.
In case, for some reasons, `fAnyoneCanPay` is set with `CSW` int tx, for CSW input the code of inputs serialization will crash because of array index is out of bounds.

Reply:
Ok, done